cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/xcm_config.rs

```diff
@@ -401,3 +431,89 @@ impl ExportXcm for BridgeHubRococoOrBridgeHubWococoSwitchExporter {
 		}
 	}
 }
+
+/// A `HandleFee` implementation that simply deposits the fees for `ExportMessage` XCM instructions
+/// into the accounts that are used for paying the relayer rewards.
+pub struct XcmExportFeeToRelayerRewardAccounts<
+	AssetTransactor,
+	DestNetwork,
+	DestParaId,
+	DestBridgeHubId,
+	BridgeLaneId,
+>(PhantomData<(AssetTransactor, DestNetwork, DestParaId, DestBridgeHubId, BridgeLaneId)>);
+
+impl<
+		AssetTransactor: TransactAsset,
+		DestNetwork: Get<NetworkId>,
+		DestParaId: Get<cumulus_primitives_core::ParaId>,
+		DestBridgeHubId: Get<ChainId>,
+		BridgeLaneId: Get<LaneId>,
+	> HandleFee
+	for XcmExportFeeToRelayerRewardAccounts<
+		AssetTransactor,
+		DestNetwork,
+		DestParaId,
+		DestBridgeHubId,
+		BridgeLaneId,
+	>
+{
+	fn handle_fee(
+		fee: MultiAssets,
+		maybe_context: Option<&XcmContext>,
+		reason: FeeReason,
+	) -> MultiAssets {
+		if matches!(reason, FeeReason::Export { network: bridged_network, destination }
+				if bridged_network == DestNetwork::get() &&
+					destination == X1(Parachain(DestParaId::get().into())))
+		{
+			let source_para_account = PayRewardFromAccount::<
+				pallet_balances::Pallet<Runtime>,
+				AccountId,
+			>::rewards_account(RewardsAccountParams::new(
+				BridgeLaneId::get(),
+				DestBridgeHubId::get(),
+				RewardsAccountOwner::ThisChain,
+			));
+
+			let dest_para_account = PayRewardFromAccount::<
+				pallet_balances::Pallet<Runtime>,
+				AccountId,
+			>::rewards_account(RewardsAccountParams::new(
+				BridgeLaneId::get(),
+				DestBridgeHubId::get(),
+				RewardsAccountOwner::BridgedChain,
+			));
+
+			for asset in fee.into_inner() {
+				match asset.fun {
+					Fungible(total_fee) => {
+						let source_fee = total_fee / 2;
+						try_deposit_fee::<AssetTransactor, _>(
+							MultiAsset { id: asset.id, fun: Fungible(source_fee) }.into(),
+							maybe_context,
+							source_para_account.clone(),
+						);
+
+						let dest_fee = total_fee - source_fee;
+						try_deposit_fee::<AssetTransactor, _>(
+							MultiAsset { id: asset.id, fun: Fungible(dest_fee) }.into(),
+							maybe_context,
+							dest_para_account.clone(),
+						);
+					},
```

Comment:
nit: maybe add a comment why we're splitting fees between the two accounts

Reply:
Done